x-pack/plugins/reporting/server/types.ts

```diff
@@ -59,45 +63,19 @@ export type ScrollConfig = ReportingConfigType['csv']['scroll'];
  * Internal Types
  */
 
-// default fn type for CreateJobFnFactory
-export type CreateJobFn<JobParamsType = BaseParams, JobPayloadType = BasePayload> = (
-  jobParams: JobParamsType,
+export type CreateJobFn = (
+  jobParams: JobParamsPDFV2,
```

Comment:
We should use generics here. 

Create Job Fn's type should start like this:
```suggestion
export type CreateJobFn<JobParamsType> = (
  jobParams: JobParamsType,
```

RunTaskFn's type should start like this:
```
export type RunTaskFn<TaskPayloadType> = (
  payload: TaskPayloadType,
```